File: bookstore_backend/src/main/java/com/bookstore/services/DashboardService.java
package com.bookstore.services;

import java.time.LocalDate;
import java.util.List;

import com.bookstore.dto.dashboard.OrderSummaryResponse;
import com.bookstore.dto.dashboard.FinancialOverviewResponse;
import com.bookstore.dto.dashboard.PaymentMethodStatsResponse;
import com.bookstore.dto.dashboard.RevenuePointResponse;
import com.bookstore.dto.dashboard.TopBookResponse;
import com.bookstore.dto.dashboard.TopCustomerResponse;

public interface DashboardService {
    OrderSummaryResponse getOrderSummary();
    List<RevenuePointResponse> getRevenueByDay(LocalDate from, LocalDate to);
    List<TopBookResponse> getTopSellingBooks(int limit);
    FinancialOverviewResponse getFinancialOverview(LocalDate from, LocalDate to);
    List<PaymentMethodStatsResponse> getPaymentMethodStats(LocalDate from, LocalDate to);
    List<TopCustomerResponse> getTopCustomers(LocalDate from, LocalDate to, int limit);
}

File: bookstore_frontend/src/features/dashboard/services/dashboardService.ts
import { apiClient } from '../../../lib/apiClient';
import type { ApiResponse } from '../../../types/api';
import type {
  OrderSummaryResponse,
  FinancialOverviewResponse,
  RevenuePointResponse,
  TopBookResponse,
  TopCustomerResponse,
  PaymentMethodStatsResponse
} from '../../../types/api/dashboard';

export const dashboardService = {
  getSummary: () =>
    apiClient.get<ApiResponse<OrderSummaryResponse>>('/api/dashboard/accountant/summary'),

  getFinancialOverview: (from: string, to: string) =>
    apiClient.get<ApiResponse<FinancialOverviewResponse>>('/api/dashboard/accountant/financial-overview', {
      params: { from, to },
    }),

  getRevenueByDay: (from: string, to: string) =>
    apiClient.get<ApiResponse<RevenuePointResponse[]>>('/api/dashboard/accountant/revenue', {
      params: { from, to },
    }),

  getTopBooks: (limit: number = 5) =>
    apiClient.get<ApiResponse<TopBookResponse[]>>('/api/dashboard/accountant/top-books', {
      params: { limit },
    }),

  getTopCustomers: (from: string, to: string, limit: number = 10) =>
    apiClient.get<ApiResponse<TopCustomerResponse[]>>('/api/dashboard/accountant/top-customers', {
      params: { from, to, limit },
    }),

  getPaymentMethodStats: (from: string, to: string) =>
    apiClient.get<ApiResponse<PaymentMethodStatsResponse[]>>('/api/dashboard/accountant/payment-methods', {
      params: { from, to },
    }),
};
